### PR TITLE
Add pd.isna, Index.isna and Index.isnull

### LINKED ISSUE #135: Missing pandas.isna and pandas.Index.isna
API reference : [pandas.isna](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.isna.html) and [pandas.Index.isna](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.isna.html)

### PR DESCRIPTION
**Pull request by fabiencelier**
I have followed this [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.isna.html#pandas.isna):

> Scalar arguments (including strings) result in a scalar boolean.
> ndarrays result in an ndarray of booleans.
> For indexes, an ndarray of booleans is returned.
> For Series and DataFrame, the same type is returned, containing booleans.

I have done multiple overloaded declaration but I am not sure about the last one with `Any` not being annotated `@overload`

I have also replaced the `pd.isnull` to follow the same rules as I believe the current one is wrong.

Closes #135

### CODE REVIEW

File: pandas-stubs/__init__.pyi [hunk 1]
```diff
@@ -30,7 +30,20 @@ def concat(
...
+@overload
+def isna(obj: Union[Index, _np.ndarray]) -> _np.ndarray[bool]: ...
+def isna(obj: Any) -> bool: ...
+@overload
+def isnull(obj: DataFrame) -> DataFrame: ...
+@overload
+def isnull(obj: Series) -> Series[bool]: ...
+@overload
+def isnull(obj: Union[Index, _np.ndarray]) -> _np.ndarray[bool]: ...
+def isnull(obj: Any) -> bool: ...
```
Comment (thomkeh) [inline pandas-stubs/__init__.pyi:46]:
```suggestion
@overload
def isnull(obj: Any) -> bool: ...
```

File: pandas-stubs/__init__.pyi [hunk 2]
```diff
@@ -30,7 +30,20 @@ def concat(
...
 def cut(arr: _np.ndarray, bins: int) -> Tuple[Union[Series, _np.ndarray], _np.ndarray]: ...
 def get_dummies(df: Union[DataFrame, Series], columns: Optional[_ListLike] = ...) -> DataFrame: ...
-def isnull(df: Union[DataFrame, Series]) -> _np.ndarray: ...
+@overload
+def isna(obj: DataFrame) -> DataFrame: ...
+@overload
+def isna(obj: Series) -> Series[bool]: ...
+@overload
+def isna(obj: Union[Index, _np.ndarray]) -> _np.ndarray[bool]: ...
+def isna(obj: Any) -> bool: ...
```
Comment (thomkeh) [inline pandas-stubs/__init__.pyi:39]:
```suggestion
@overload
def isna(obj: Any) -> bool: ...
```

**Comment by fabiencelier** [comment]
Looks like there is a conflict if I annotate `@overload` the function with Any.

```
/opt/hostedtoolcache/Python/3.6.11/x64/lib/python3.6/site-packages/pandas-stubs/__init__.pyi:34: error: Overloaded function signatures 1 and 4 overlap with incompatible return types
/opt/hostedtoolcache/Python/3.6.11/x64/lib/python3.6/site-packages/pandas-stubs/__init__.pyi:36: error: Overloaded function signatures 2 and 4 overlap with incompatible return types
```

I will investigate tomorrow

**Comment by olliethomas** [comment]
oooh, I think black has been updated since this PR was made

**Comment by thomkeh** [comment]
We had to remove the `Any` overload for now, because it gave a mypy error. We can revisit this if you need it.

**Pull request merged**